api/app/signals/apps/history/services/history_log.py

```diff
@@ -1,24 +1,22 @@
 # SPDX-License-Identifier: MPL-2.0
 # Copyright (C) 2022 Gemeente Amsterdam
-import threading
-
+from django.core.handlers.wsgi import WSGIRequest
 from django.utils import timezone
 
 from signals.apps.history.models import Log
 
 
 class HistoryLogService:
-    thread = threading.local()
 
     @staticmethod
-    def log_update(instance) -> None:
+    def log_update(request: WSGIRequest, instance) -> None:
         if not hasattr(instance, 'changed_data') or not callable(instance.changed_data):
             return
 
         changed_data = instance.changed_data()
         if changed_data:
-            if HistoryLogService.thread.request and HistoryLogService.thread.request.user:
-                created_by = HistoryLogService.thread.request.user.username
+            if request and request.user:
```

Comment:
Change to user. And check if the user is given and if it is not the anonymous user.